ClassNotFoundException for beanparam class with generics in external artifact

### Describe the bug

In my app I want to put some of the commonly used classes in a common lib.
This however causes a ClassNotFoundException from resteasy-reactive (see below) for this constellation:

These 2 classes are in my main quarkus app.

Resource
```
@GET
    @Produces(MediaType.TEXT_PLAIN)
    public String hello(@BeanParam GreetingListFilter<GreetingListSortAttribute> listFilter) {
        return listFilter.sortAttribute.name();
    }
```

```
public enum GreetingListSortAttribute implements SortAttribute{
    FORMALITY_FACTOR,
    LANGUAGE;

}
```

This one (and SortAttribute interface) is in an external lib:
```
public class GreetingListFilter<T extends SortAttribute> {
    @QueryParam("sortAttribute")
    // bogus default value so that rr calls fromString
    @DefaultValue("-")
    public T sortAttribute;
}
```

If I have the GreetingListFilter in the same module as app, everything works. It also works if the lib project is a sister module adjacent to the app.
It does not work if the lib is a fully external artifact.
I have added a META-INF/beans.xml in the lib for good measure, does not help. 

### Expected behavior

App starts


As endgoal, I want to call `http://localhost:8080/hello?sortAttribute=LANGUAGE`, and have the sortAttribute of GreetingListFilter be an instance of GreetingListSortAttribute. This works already if the lib is not external.

### Actual behavior

App does not start
```
2024-03-20 16:58:45,853 ERROR [io.qua.run.boo.StartupActionImpl] (Quarkus Main Thread) Error running Quarkus: java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:118)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at io.quarkus.runner.bootstrap.StartupActionImpl$1.run(StartupActionImpl.java:113)
        at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.lang.ExceptionInInitializerError
        at java.base/jdk.internal.misc.Unsafe.ensureClassInitialized0(Native Method)
        at java.base/jdk.internal.misc.Unsafe.ensureClassInitialized(Unsafe.java:1160)
        at java.base/jdk.internal.reflect.MethodHandleAccessorFactory.ensureClassInitialized(MethodHandleAccessorFactory.java:300)
        at java.base/jdk.internal.reflect.MethodHandleAccessorFactory.newConstructorAccessor(MethodHandleAccessorFactory.java:103)
        at java.base/jdk.internal.reflect.ReflectionFactory.newConstructorAccessor(ReflectionFactory.java:200)
        at java.base/java.lang.reflect.Constructor.acquireConstructorAccessor(Constructor.java:549)
        at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:486)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:70)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:44)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:124)
        at io.quarkus.runner.GeneratedMain.main(Unknown Source)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        ... 3 more
Caused by: java.lang.RuntimeException: Failed to start quarkus
        at io.quarkus.runner.ApplicationImpl.<clinit>(Unknown Source)
        ... 16 more
Caused by: java.lang.NoClassDefFoundError: org/acme/GreetingListSortAttribute$quarkusrestparamConverter$
        at org.acme.GreetingListFilter.__quarkus_init_converter__sortAttribute(GreetingListFilter.java)
        at io.quarkus.rest.runtime.__QuarkusInit.init(Unknown Source)
        at io.quarkus.resteasy.reactive.server.runtime.ResteasyReactiveRecorder.createDeployment(ResteasyReactiveRecorder.java:144)
        at io.quarkus.deployment.steps.ResteasyReactiveProcessor$setupDeployment713137389.deploy_0(Unknown Source)
        at io.quarkus.deployment.steps.ResteasyReactiveProcessor$setupDeployment713137389.deploy(Unknown Source)
        ... 17 more
Caused by: java.lang.ClassNotFoundException: org.acme.GreetingListSortAttribute$quarkusrestparamConverter$
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
        at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:518)
        at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:468)
        ... 22 more


2024-03-20 16:58:45,871 INFO  [io.qua.dep.dev.IsolatedDevModeMain] (main) Attempting to start live reload endpoint to recover from previous Quarkus startup failure

```

### How to Reproduce?

Reproducer: 
[rr-gen-type.zip](https://github.com/quarkusio/quarkus/files/14669565/rr-gen-type.zip)

1. cd lib
2. mvn clean install
3. cd ..
4. mvn clean compile quarkus:dev
5. error from above, does not start
6. ctrl+c, now in the pom.xml of the parent, include lib as a module
7. mvn clean compile quarkus:dev
8. app starts

### Output of `uname -a` or `ver`

MINGW64_NT-10.0-19045 NANBCHL9NG3 3.3.6-341.x86_64 2022-09-05 20:28 UTC x86_64 Msys

### Output of `java -version`

openjdk 21.0.2 2024-01-16 LTS OpenJDK Runtime Environment Temurin-21.0.2+13 (build 21.0.2+13-LTS) OpenJDK 64-Bit Server VM Temurin-21.0.2+13 (build 21.0.2+13-LTS, mixed mode, sharing)

### Quarkus version or git rev

3.8.3

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.4 (9b656c72d54e5bacbed989b64718c159fe39b537) Maven home: C:\eclipse\tools\java\maven Java version: 21.0.2, vendor: Eclipse Adoptium, runtime: C:\eclipse\tools\java\21 Default locale: de_DE, platform encoding: UTF-8 OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"

### Additional information

_No response_

--- a/extensions/resteasy-reactive/rest/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/ResteasyReactiveProcessor.java
+++ b/extensions/resteasy-reactive/rest/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/ResteasyReactiveProcessor.java
@@ -500,8 +500,7 @@ public class ResteasyReactiveProcessor {
                                         initConverters.getMethodParam(0));
                             }))
                     .setConverterSupplierIndexerExtension(new GeneratedConverterIndexerExtension(
-                            (name) -> new GeneratedClassGizmoAdaptor(generatedClassBuildItemBuildProducer,
-                                    applicationClassPredicate.test(name))))
+                            (name) -> new GeneratedClassGizmoAdaptor(generatedClassBuildItemBuildProducer, true)))
                     .setHasRuntimeConverters(!paramConverterProviders.getParamConverterProviders().isEmpty())
                     .setClassLevelExceptionMappers(
                             classLevelExceptionMappers.isPresent() ? classLevelExceptionMappers.get().getMappers()

